Makefile

```diff
@@ -14,5 +31,100 @@ vet:
 test:
 	go test -v ./...
 
-# Default target
-all: build test
+# Packit helpers
+.PHONY: vendor-tarball packit-create-archive vendor-licenses
+
+vendor-tarball: $(VENDOR_TARBALL)
+
+$(VENDOR_TARBALL):
+	mkdir -p "$(SOURCEDIR)"
+	GOTOOLCHAIN=local go_vendor_archive create --config "$(GO_VENDOR_TOOLS_FILE)" .
+	@mv -f vendor.tar.bz2 "$(VENDOR_TARBALL)"
+
+packit-create-archive: $(SOURCE_TARBALL) $(VENDOR_TARBALL)
+	@ls -1 "$(SOURCE_TARBALL)" | head -n1
+
+$(SOURCE_TARBALL):
+	mkdir -p "$(SOURCEDIR)"
+	git archive --format=tar --prefix="$(PROJECT)-$(VERSION)/" HEAD | gzip > "$(SOURCE_TARBALL)"
+
+vendor-licenses:
+	go_vendor_license --config "$(GO_VENDOR_TOOLS_FILE)" .
+
+#
+# Building packages
+#
+# The following rules build FDO packages from the current HEAD commit,
+# based on the spec file in build/package/rpm directory. The resulting packages
+# have the commit hash in their version, so that they don't get overwritten when calling
+# `make rpm` again after switching to another branch or adding new commits.
+#
+# All resulting files (spec files, source rpms, rpms) are written into
+# ./rpmbuild, using rpmbuild's usual directory structure (in lowercase).
+#
+
+RPM_BASE_DIR           := $(CURDIR)/build/package/rpm
+SPEC_FILE_NAME         := $(PROJECT).spec
+SPEC_FILE              := $(RPM_BASE_DIR)/$(SPEC_FILE_NAME)
+
+RPMBUILD_TOP_DIR       := $(CURDIR)/rpmbuild
+RPMBUILD_BUILD_DIR     := $(RPMBUILD_TOP_DIR)/build
+RPMBUILD_RPMS_DIR      := $(RPMBUILD_TOP_DIR)/rpms
+RPMBUILD_SPECS_DIR     := $(RPMBUILD_TOP_DIR)/specs
+RPMBUILD_SOURCES_DIR   := $(RPMBUILD_TOP_DIR)/sources
+RPMBUILD_SRPMS_DIR     := $(RPMBUILD_TOP_DIR)/srpms
+RPMBUILD_BUILDROOT_DIR := $(RPMBUILD_TOP_DIR)/buildroot
+
+RPMBUILD_GOLANG_VENDOR_TOOLS_FILE := $(RPMBUILD_SOURCES_DIR)/$(GO_VENDOR_TOOLS_FILE_NAME)
+RPMBUILD_SPECFILE                 := $(RPMBUILD_SPECS_DIR)/$(PROJECT)-$(VERSION).spec
+RPMBUILD_TARBALL                  := $(RPMBUILD_SOURCES_DIR)/$(PROJECT)-$(VERSION).tar.gz
+RPMBUILD_VENDOR_TARBALL           := $(RPMBUILD_SOURCES_DIR)/$(PROJECT)-$(VERSION)-vendor.tar.bz2
```

Comment:
My obsessive-compulsive disorder is triggering here and asking for coherence: would it be possible to use the same compression for all the sources?
In the `go-fdo-server` repo I verified the difference in size between the `tar.gz` and `tar.bz2` files were close to 0 so maybe we can use `tar.gz` both of them as it's simpler.